src/containers/mappage/parsingmethods/districtParsing.js

```diff
@@ -1,148 +1,44 @@
-    return "No Reports";
-  } else {
-    const nonShortages = DistrictArrDateRange.map(
-      ({ nonShortagesReported }) => nonShortagesReported
-    ).reduce((a, b) => a + b, 0);
-    return nonShortages;
-  }
-};
-
-export const currentDistricTestingToDate = (
-  filteredReportData,
-  currentDistrict,
-  requestedReport
-) => {
-  const currentDistrictArr = filteredReportData.filter(
-    (districtObj) => districtObj.district === currentDistrict.district
-  );
-  const DistrictArrDateRange = currentDistrictArr.filter((districtObj) =>
-    moment(districtObj.reported_date).isSameOrBefore(
-      requestedReport.reportedDate,
-      "day"
-    )
-  );
-  if (DistrictArrDateRange.length === 0) {
-    return "No Reports";
-  } else {
-    const testing = DistrictArrDateRange.map(
-      ({ testingUnavailable }) => testingUnavailable
-    ).reduce((a, b) => a + b, 0);
-    return testing;
-  }
-};
-
-export const currentDateDistrictShortages = (
-  filteredReportData,
-  currentDistrict,
-  requestedReport
-) => {
-  const currentDistrictArr = filteredReportData.filter(
-    (districtObj) => districtObj.district === currentDistrict.district
-  );
-  const selectedDateDistrictArr = currentDistrictArr.filter(
-    (districtObj) => districtObj.reported_date === requestedReport.reportedDate
-  );
-  if (selectedDateDistrictArr.length === 0) {
-    return "No Reports";
-  } else {
-    const shortages = selectedDateDistrictArr
-      .map(({ shortagesReported }) => shortagesReported)
-      .reduce((a, b) => a + b, 0);
-    return shortages;
-  }
-};
-
-export const currentDateDistrictNonShortages = (
-  filteredReportData,
-  currentDistrict,
-  requestedReport
-) => {
-  const currentDistrictArr = filteredReportData.filter(
-    (districtObj) => districtObj.district === currentDistrict.district
-  );
-  const selectedDateDistrictArr = currentDistrictArr.filter(
-    (districtObj) => districtObj.reported_date === requestedReport.reportedDate
-  );
-  if (selectedDateDistrictArr.length === 0) {
-    return "No Reports";
-  } else {
-    const shortages = selectedDateDistrictArr
-      .map(({ nonShortagesReported }) => nonShortagesReported)
-      .reduce((a, b) => a + b, 0);
-    return shortages;
-  }
-};
-
-export const currentDateDistrictTesting = (
-  filteredReportData,
-  currentDistrict,
-  requestedReport
-) => {
-  const currentDistrictArr = filteredReportData.filter(
-    (districtObj) => districtObj.district === currentDistrict.district
-  );
-  const selectedDateDistrictArr = currentDistrictArr.filter(
-    (districtObj) => districtObj.reported_date === requestedReport.reportedDate
   );
-  if (selectedDateDistrictArr.length === 0) {
-    return "No Reports";
-  } else {
-    const shortages = selectedDateDistrictArr
-      .map(({ testingUnavailable }) => testingUnavailable)
-      .reduce((a, b) => a + b, 0);
-    return shortages;
-  }
+  const totals = filteredDistricts
+    .map(({ citedShortage, citedNoTesting, reports }) => {
```

Comment:
Why map over this?  Just to eliminate fields from the objects in the array?

Reply:
**citedShortage, citedNotesting**, and **reports** are top level fields within the array objects. **shortages** and **testData** are deeper obects, so the map is to isolate the parts of the array  to reduce the properties within the deeper objects.